### PR TITLE
Merge IMOS-Checker-Issue-3 into IMOS-Checker (fixes #3)

### LINKED ISSUE #3: Missing IMOS check for Conventions global attribute
According to the xls spreadsheet this has been implemented, but looking at the latest commit (f70671855973429f0baa658e98674ceba57b93bd) this doesn't seem to be the case. This is a high priority check.

### PR DESCRIPTION
**Pull request by mhidas**
@fxmzb123 I tried to merge this but there's a conflict with the test file imos_bad_data.nc, which seems to have changed in both branches. Are you able to resolve this?

### CODE REVIEW

**Comment by fxmzb123** [comment]
This has been resolved. You can merge it now.

**Comment by mhidas** [comment]
@fxmzb123 I just tried running the IMOS unit tests on this branch, and all of them result in an error like this:

```
Traceback (most recent call last):
  File "compliance_checker/tests/test_imos.py", line 47, in setUp
    self.bad_dataset = self.get_pair(static_files['bad_data'])
  File "compliance_checker/tests/test_imos.py", line 57, in get_pair
    nc_dataset = Dataset(nc_dataset, 'r')
  File "netCDF4/_netCDF4.pyx", line 1547, in netCDF4._netCDF4.Dataset.__init__ (netCDF4/_netCDF4.c:9551)
RuntimeError: NetCDF: HDF error
```

It looks like the test file imos_bad_data.nc is corrupted or incorrect format. I get the same `HDF error` if I run `ncdump` on it.

File: compliance_checker/tests/test_imos.py [hunk 1]
```diff
@@ -98,6 +98,39 @@ def test_check_project_attribute(self):
...
+
+        for result in ret_val:
+            self.assertTrue(result.value)
+
+        ret_val = self.imos.check_conventions(self.bad_dataset)
+        
+        for result in ret_val:
+            self.assertFalse(result.value)
+
+    def test_check_conventions_attribute(self):
```
Comment (mhidas) [inline compliance_checker/tests/test_imos.py:113]: Why is this def in here three times?

File: compliance_checker/tests/test_imos.py [hunk 2]
```diff
@@ -429,6 +462,15 @@ def test_check_quality_control_variable_listed(self):
...
         self.assertTrue(ret_val[1].value)
+        
+    def test_check_quality_control_conventions_for_quality_control_variable(self):
+        self.imos.setup(self.test_variable_dataset)
+        ret_val = self.imos.check_quality_control_conventions_for_quality_control_variable(self.test_variable_dataset)
+
+        self.assertTrue(ret_val != None)
+        self.assertTrue(len(ret_val) == 1)
+
+        self.assertTrue(ret_val[0].value)
```
Comment (mhidas) [inline compliance_checker/tests/test_imos.py:473]: We should also test that the check fails when it should fail.

**Comment by mhidas** [comment]
Don't need this now. Fix was done in #22 .

**Pull request closed without merging**